vllm/model_executor/models/adapters.py

```diff
@@ -0,0 +1,92 @@
+from collections.abc import Iterable
+from typing import Any, TypeVar
+
+import torch
+import torch.nn as nn
+
+from .interfaces_base import VllmModelForEmbedding, is_embedding_model
+
+_T = TypeVar("_T", bound=type[nn.Module])
+
+
+def for_embedding(cls: _T) -> _T:
+    """Subclass an existing vLLM model to support embeddings."""
```

Comment:
```suggestion
def embedding_model_wrapper(cls: _T) -> _T:
    """Wrapper function to convert an existing vLLM model to embedding model"""
```
I think `for_embedding` might be a little bit confusing. We can use a clearer name for this.

Reply:
Renamed it to `as_embedding_model`. I'm hesitant to use "wrapper" because it doesn't really wrap the original class - instead it creates a new subclass.